### PR TITLE
feat(dependabot): Add dependabot to check for vulnerabilies and updat…

### PR DESCRIPTION
**Pull request by BLANKatGITHUB**
This will automatically scan for updates in dependencies and create pull request to merge them.

* Added daily update schedule for GitHub Actions workflows in the root directory.
* Added daily update schedule for Go modules in the `/contrib/charts/dragonfly` directory. 
* Added daily update schedule for Python packages in the `/tests/dragonfly` directory. 
* Added daily update schedule for Go modules in the `/tools/replay` directory. 
* Added daily update schedule for Python packages in the `/tools` directory.

### CODE REVIEW

File: .github/dependabot.yml [hunk 1]
```diff
@@ -0,0 +1,26 @@
+version: 2
+updates:
+  - package-ecosystem: github-actions
+    directory: /
+    schedule:
+      interval: daily
+
+  - package-ecosystem: gomod
```
Comment (romange) [inline .github/dependabot.yml:8]: let's group all directories for each package ecosystem together.
See https://docs.github.com/en/code-security/dependabot/dependabot-version-updates/configuration-options-for-the-dependabot.yml-file#directories for more details.

**Comment by romange** [comment]
Weekly is enough, thanks! 🙏🏼

**Comment by BLANKatGITHUB** [comment]
well I made adjustments , it will check for latest versions of dependencies , you will have to change the version of languages manually though for python and go .  Dependabot  makes pr for dependencies not language .

**Review by romange — APPROVED** [review]

**Comment by romange** [comment]
Thanks!

**Comment by romange** [comment]
@BLANKatGITHUB dependabot start working but it is a bit noisy :)

I saw it is possible to group updates into a single update PR using groups:
https://docs.github.com/en/code-security/dependabot/dependabot-version-updates/configuration-options-for-the-dependabot.yml-file#example-2

do you mind improving the file so that it will group the updates under a single `dev-dependecies` group?

**Comment by BLANKatGITHUB** [comment]
I will try.

**Comment by romange** [comment]
Also, worth limiting these updates to (security) patches only if possible. We do not always want to update to latest due to the distribution limitations.

**Comment by BLANKatGITHUB** [comment]
umm I dont understand? You want depandabot to create only security update prs?

**Comment by BLANKatGITHUB** [comment]
So basically only security updates right?

**Comment by romange** [comment]
yes, if possible

**Pull request merged**